src/main.js

```diff
@@ -162,6 +162,20 @@ class PercyClient {
     return this._httpGet(`${this.apiUrl}/builds/${buildId}`);
   }
 
+  // This method is unavailable to normal write-only project tokens.
+  getBuilds(project, filter) {
+    filter = filter || {};
+    let queryString = Object.keys(filter)
+      .map(key => 'filter[' + key + ']=' + filter[key])
+      .join('&');
```

Comment:
Optionally take a look at https://nodejs.org/api/querystring.html if you haven't already. It might come in handy here. There's some nice escaping and object to query string features.